File: TelerikUIExamplesInCSharp/Examples/SideDrawer/TransitionsExample/SideDrawerTransitionsModalController.cs
using System;

using ObjCRuntime;
using Foundation;
using UIKit;
using CoreAnimation;
using CoreGraphics;

using TelerikUI;

namespace Examples
{
	public class SideDrawerTransitionsModalController : SideDrawerGettingStartedModalController
	{
		public override void ViewDidLoad ()
		{
			base.ViewDidLoad ();
			this.AddButtons ();
		}

		public virtual void AddButtons ()
		{
			this.CreateButton ("Push", this, new Selector ("PushTransition"), new CGPoint (15, 80));
			this.CreateButton ("Reveal", this, new Selector ("RevealTransition"), new CGPoint (15, 130));
			this.CreateButton ("Reverse Slide Out", this, new Selector ("ReverseSlideOutTransition"), new CGPoint (15, 180));
			this.CreateButton ("Slide Along", this, new Selector ("SlideAlongTransition"), new CGPoint (15, 230));
			this.CreateButton ("Slide In On Top", this, new Selector ("SlideInOnTopTransition"), new CGPoint (15, 280));
			this.CreateButton ("Scale Up", this, new Selector ("ScaleUpTransition"), new CGPoint (15, 330));
			this.CreateButton ("Fade In", this, new Selector ("FadeInTransition"), new CGPoint (15, 380));
		}

		[Export ("PushTransition")]
		private void PushTransition() 
		{
			this.SideDrawer.Transition = TKSideDrawerTransitionType.Push;
			this.SideDrawer.Fill = new TKSolidFill (UIColor.Gray);
			this.SideDrawer.HeaderView = new SideDrawerHeader (false, null, null);
			this.SideDrawer.Show ();
		}

		[Export ("RevealTransition")]
		private void RevealTransition()
		{
			this.SideDrawer.Transition = TKSideDrawerTransitionType.Reveal;
			this.SideDrawer.Fill = new TKSolidFill (UIColor.Gray);
			this.SideDrawer.HeaderView = new SideDrawerHeader (false, null, null);
			this.SideDrawer.Show ();
		}

		[Export ("ReverseSlideOutTransition")]
		private void ReverseSlideOutTransition()
		{
			this.SideDrawer.Transition = TKSideDrawerTransitionType.ReverseSlideOut;
			this.SideDrawer.Fill = new TKSolidFill (UIColor.Gray);
			this.SideDrawer.HeaderView = new SideDrawerHeader (false, null, null);
			this.SideDrawer.Show ();
		}

		[Export ("SlideAlongTransition")]
		private void SlideAlongTransition()
		{
			this.SideDrawer.Transition = TKSideDrawerTransitionType.SlideAlong;
			this.SideDrawer.Fill = new TKSolidFill (UIColor.Gray);
			this.SideDrawer.HeaderView = new SideDrawerHeader (false, null, null);
			this.SideDrawer.Show ();
		}

		[Export ("SlideInOnTopTransition")]
		private void SlideInOnTopTransition()
		{
			this.SideDrawer.Transition = TKSideDrawerTransitionType.SlideInOnTop;
			this.SideDrawer.Fill = new TKSolidFill (UIColor.Clear);
			this.SideDrawer.HeaderView = new SideDrawerHeader (true, this, new Selector("DismissSideDrawer"));
			this.SideDrawer.Show ();
		}

		[Export ("ScaleUpTransition")]
		private void ScaleUpTransition()
		{
			this.SideDrawer.Transition = TKSideDrawerTransitionType.ScaleUp;
			this.SideDrawer.Fill = new TKSolidFill (UIColor.Gray);
			this.SideDrawer.HeaderView = new SideDrawerHeader (false, null, null);
			this.SideDrawer.Show ();
		}

		[Export ("FadeInTransition")]
		private void FadeInTransition()
		{
			this.SideDrawer.Transition = TKSideDrawerTransitionType.FadeIn;
			this.SideDrawer.Fill = new TKSolidFill (UIColor.Gray);
			this.SideDrawer.HeaderView = new SideDrawerHeader (true, this, new Selector("DismissSideDrawer"));
			this.SideDrawer.Show ();
		}

		public void CreateButton(string title, NSObject target, Selector selector, CGPoint origin)
		{
			NSString titleString = new NSString (title);
			UIStringAttributes attributes = new UIStringAttributes () { Font = UIFont.SystemFontOfSize (18) };
			CGSize titleSize = titleString.GetSizeUsingAttributes(attributes);
			UIButton button = new UIButton (new CGRect (origin.X, origin.Y, titleSize.Width, 44));
			button.TitleLabel.Font = UIFont.SystemFontOfSize (14);
			button.Layer.BorderWidth = 1.0f;
			button.Layer.BorderColor = UIColor.White.CGColor;
			button.Layer.CornerRadius = 3.0f;
			button.SetTitle (title, UIControlState.Normal);
			button.SetTitleColor (UIColor.White, UIControlState.Normal);
			button.AddTarget (target, selector, UIControlEvent.TouchUpInside);
			this.View.AddSubview (button);
		}
	}
}

File: TelerikUIExamplesInSwift/TelerikUIExamplesInSwift/SideDrawerTransitionsModalController.swift
//
//  SideDrawerTransitionsModalController.swift
//  TelerikUIExamplesInSwift
//
//  Copyright (c) 2015 Telerik. All rights reserved.
//

class SideDrawerTransitionsModalController: SideDrawerGettingStartedModalController, TKSideDrawerDelegate {

    override func viewDidLoad() {
        super.viewDidLoad()
        
        self.createButton("Push", target: self, selector: "pushTransition", origin: CGPointMake(15, 80))
        self.createButton("Reveal", target: self, selector: "revealTransition", origin: CGPointMake(15, 130))
        self.createButton("Reverse Slide Out", target: self, selector: "reverseSlideOutTransition", origin: CGPointMake(15, 180))
        self.createButton("Slide Along", target: self, selector: "slideAlongTransition", origin: CGPointMake(15, 230))
        self.createButton("Slide In On Top", target: self, selector: "slideInOnTopTransition", origin: CGPointMake(15, 280))
        self.createButton("Scale Up", target: self, selector: "scaleUpTransition", origin: CGPointMake(15, 330))
        self.createButton("Fade In", target: self, selector: "fadeInTransition", origin: CGPointMake(15, 380))
    }
    
    func createButton(title: String, target: AnyObject, selector: Selector, origin: CGPoint) {
        let titleString: NSString = title as NSString
        let titleSize = titleString.sizeWithAttributes([NSFontAttributeName : UIFont.systemFontOfSize(18.0)])
        let button = UIButton(frame: CGRectMake(origin.x, origin.y, titleSize.width, 44))
        button.titleLabel?.font = UIFont.systemFontOfSize(14)
        button.layer.borderWidth = 1.0
        button.layer.borderColor = UIColor.whiteColor().CGColor
        button.layer.cornerRadius = 3.0
        button.setTitle(title, forState: UIControlState.Normal)
        button.setTitleColor(UIColor.whiteColor(), forState: UIControlState.Normal)
        button.addTarget(self, action: selector, forControlEvents: UIControlEvents.TouchUpInside)
        self.view.addSubview(button)
    }
    
    func pushTransition() {
        let sideDrawer = self.sideDrawer
        sideDrawer.transition = TKSideDrawerTransitionType.Push
        sideDrawer.fill = TKSolidFill(color: UIColor.grayColor())
        sideDrawer.headerView = SideDrawerHeaderView(addButton: false, target: nil, selector: nil)
        self.showSideDrawer()
    }
    
    func revealTransition() {
        let sideDrawer = self.sideDrawer
        sideDrawer.transition = TKSideDrawerTransitionType.Reveal
        sideDrawer.fill = TKSolidFill(color: UIColor.grayColor())
        sideDrawer.headerView = SideDrawerHeaderView(addButton: false, target: nil, selector: nil)
        self.showSideDrawer()
    }
    
    func reverseSlideOutTransition() {
        let sideDrawer = self.sideDrawer
        sideDrawer.transition = TKSideDrawerTransitionType.ReverseSlideOut
        sideDrawer.fill = TKSolidFill(color: UIColor.grayColor())
        sideDrawer.headerView = SideDrawerHeaderView(addButton: false, target: nil, selector: nil)
        self.showSideDrawer()
    }
    
    func slideAlongTransition() {
        let sideDrawer = self.sideDrawer
        sideDrawer.transition = TKSideDrawerTransitionType.SlideAlong
        sideDrawer.fill = TKSolidFill(color: UIColor.grayColor())
        sideDrawer.headerView = SideDrawerHeaderView(addButton: false, target: nil, selector: nil)
        self.showSideDrawer()
    }
    
    func slideInOnTopTransition() {
        let sideDrawer = self.sideDrawer
        sideDrawer.transition = TKSideDrawerTransitionType.SlideInOnTop
        sideDrawer.fill = TKSolidFill(color: UIColor.clearColor())
        sideDrawer.headerView = SideDrawerHeaderView(addButton: true, target: self, selector: Selector("dismissSideDrawer"))
        self.showSideDrawer()
    }
    
    func scaleUpTransition() {
        let sideDrawer = self.sideDrawer
        sideDrawer.transition = TKSideDrawerTransitionType.ScaleUp
        sideDrawer.fill = TKSolidFill(color: UIColor.grayColor())
        sideDrawer.headerView = SideDrawerHeaderView(addButton: false, target: nil, selector: nil)
        self.showSideDrawer()
    }
    
    func fadeInTransition() {
        let sideDrawer = self.sideDrawer
        sideDrawer.transition = TKSideDrawerTransitionType.FadeIn
        sideDrawer.fill = TKSolidFill(color: UIColor.grayColor())
        sideDrawer.headerView = SideDrawerHeaderView(addButton: true, target: self, selector: Selector("dismissSideDrawer"))
        self.showSideDrawer()
    }
}
